### PR TITLE
Adding JDG tutorial + code corrections

### PR DESCRIPTION
**Pull request by mbogoevici**
Pull request for reviewing the tutorial

### CODE REVIEW

File: tutorial/DataGrid.asciidoc [hunk 1]
```diff
@@ -0,0 +1,1499 @@
...
+Adding Infinispan
+-----------------
+
+First, you need to decide how you will use Infinispan in the project. You can opt between two access patterns:
+
+Library ::
+    The data grid nodes are embedded in the application. In this case, we need to add the core data grid libraries as a dependency to the project.
+
+Remote client-server ::
+    The data grid nodes are started separately and accessed through a client library. It is this library that will be added as a dependency.
```
Comment (pmuir) [inline tutorial/DataGrid.asciidoc:56]: "It is this library that" -> "Only the client library is added as a dependency".

File: tutorial/DataGrid.asciidoc [hunk 2]
```diff
@@ -0,0 +1,1499 @@
...
+
+First, you need to decide how you will use Infinispan in the project. You can opt between two access patterns:
+
+Library ::
+    The data grid nodes are embedded in the application. In this case, we need to add the core data grid libraries as a dependency to the project.
+
+Remote client-server ::
+    The data grid nodes are started separately and accessed through a client library. It is this library that will be added as a dependency. 
+
+For TicketMonster, we will use the library access pattern, as in this particular case we can benefit from the simpler setup. For a more detailed description of the pros and cons of each access pattern, you can read a more detailed explanation in the project documentation. In any case, switching from one mode to the other is non-intrusive, the only major difference being the infrastructure setup.
```
Comment (pmuir) [inline tutorial/DataGrid.asciidoc:58]: A link to the detailed pros and cons would be good.

File: tutorial/DataGrid.asciidoc [hunk 3]
```diff
@@ -0,0 +1,1499 @@
...
+@ApplicationScoped
+public class CacheProducer {
+
+    @Produces
+    @ApplicationScoped
+    public EmbeddedCacheManager getCacheContainer() {
+        GlobalConfiguration glob = new GlobalConfigurationBuilder()
+                .nonClusteredDefault() //Helper method that gets you a default constructed GlobalConfiguration, preconfigured for use in LOCAL mode
+                .globalJmxStatistics().enable() //This method allows enables the jmx statistics of the global configuration.
+                .transport().defaultTransport()
```
Comment (pmuir) [inline tutorial/DataGrid.asciidoc:125]: This needs removing, you set up a non clustered default, but then enable a transport. This makes this not a local cache.

File: tutorial/DataGrid.asciidoc [hunk 4]
```diff
@@ -0,0 +1,1499 @@
...
+     *
+     * @param cart - the cart to be removed
+     */
+    public void delete(Cart cart) {
+        this.cartsCache.remove(cart.getId());
+    }
+}
+---------------------------------------------------------------------------------
+
+Now you can go on and implement the RESTful service for managing carts.
```
Comment (pmuir) [inline tutorial/DataGrid.asciidoc:732]: You've suddenly switched style here from we -> you

**Comment by pmuir** [comment]
This looks great. @vineetreynolds - please review!

**Comment by VineetReynolds** [comment]
@pmuir @mbogoevici I'm letting you know that I've been reviewing this. The changes apply cleanly on the current master. But breaks the app - at the very least the functionality to book tickets is broken at the presentation layer.

I suspect the CartService changes since the ticket-summary-view Underscore template fails to render (properties like section in the TicketPrice object is not available on the client). I'll attempt to correct this issue and the rest pointed out by Pete.

**Comment by VineetReynolds** [comment]
@mbogoevici Thanks for the answer. I reverted those changes, and when doing so, I think I hit on the rationale on why you removed the TicketReservationRequest class - the client could do away with submitting TicketRequests. It would have been fine, except that the Cart needed to store additional info.

I've submitted some fixes on top of your work as a new PR (#27).

**Comment by VineetReynolds** [comment]
@mbogoevici @pmuir This has been merged: https://github.com/jboss-jdf/ticket-monster/pull/27

**Pull request closed without merging**